File: 806.写字符串需要的行数.go
/*
 * @lc app=leetcode.cn id=806 lang=golang
 *
 * [806] 写字符串需要的行数
 */

// @lc code=start
func numberOfLines(widths []int, s string) []int {
	currLength, needLines := 0, 0
	for i := 0; i < len(s); i++ {
		curr := widths[s[i]-'a']
		if curr+currLength > 100 {
			needLines++
			currLength = curr
		} else {
			currLength += curr
		}
	}
	return []int{needLines + 1, currLength}
}

// @lc code=end

File: 806.写字符串需要的行数.py
#
# @lc app=leetcode.cn id=806 lang=python3
#
# [806] 写字符串需要的行数
#

# @lc code=start
class Solution:
    def numberOfLines(self, widths: List[int], s: str) -> List[int]:
        curr_length, lines = 0, 0
        ord_a = ord("a")
        for letter in s:
            curr = widths[ord(letter) - ord_a]
            if curr_length + curr > 100:
                lines += 1
                curr_length = curr
            else:
                curr_length += curr
        return [lines + 1, curr_length]
# @lc code=end

